documentation/src/main/asciidoc/topics/proc_displaying-metrics-st-operation.adoc

```diff
@@ -0,0 +1,41 @@
+[id='displaying-metrics-st-operationadoc_{context}']
+= Displaying metrics during a state transfer operation
+
+You can export time metrics for clustered caches that {brandname} redistributes across nodes. 
+
+A state transfer operation occurs when a clustered cache topology changes, such as a node joining a cluster, so the cache's data must be redistributed to all nodes in a cluster. 
+
+[NOTE]
+====
+You cannot perform a state transfer operation in invalidation mode, because this mode restricts database write operations after state changes occur. 
+====
+
+{brandname} generates time metrics that are compatible with the REST API and the JMX API. 
+
+Consider accessing metrics through the REST endpoint. 
+This endpoint uses a `StateTransferManager` attribute to query the following methods:
+
+* `getInflightSegmentTransferCount()`, which returns the number of in-flight requested segments.
+* `getInflightTransactionalSegmentCount()`, which returns the number of in-flight transactional requested segments.
```

Comment:
As I said in Gdocs, I think this may cause confusion. We can leave these details out and explain stuff in the Procedure section.

Reply:
Remove these points completely? A procedure should only cover actionable steps. These bullet points are more informative, so would fit better in an intro section, IMO.

Reply:
Yes. AFAIK this stuff are not public to a direct access, I think it would be better to avoid referencing it directly by name and methods. We can rephrase this, though. There exists a few more things exposed from the state transfer that we could list. [Here](https://docs.jboss.org/infinispan/14.0/apidocs/jmxComponents.html#StateTransferManager) we list what is exposed with an associated description, we could use this instead. We list what we expose without referencing how it is exposed yet and without referencing internals. The next section then describe how to collect the values.

Reply:
Thanks, @jabolina .

I removed these bullet points and added the following sentence (link to attributes in the additional resources section):

> The StateTransafer Manager component uses specific attributes to handle a state transfer operation.

WDYT?

Reply:
Maybe we can go without referring to `StateTransferManager`. I think that we could keep the line that says that we expose metrics for each cache, and, in the sequence, we either list what we expose using the bullets or only say that the link in the additional resources has a list. WDYT?

If we are to take it literally, the attributes are exposed, we do not use them to handle the operation. Not sure if we go to such lengths, but maybe it could lead to confusion in the future.

Reply:
Thanks, @jabolina . I removed that sentence and listed the attributes. Would the correct statement be correct?

> A state transfer exposes the following attributes as properties, so that {brandname} can export metrics from each cache:

Reply:
Maybe we use the descriptions instead of the name. Do you think we should add these or are we going out of scope?

Reply:
Hi @jabolina .

Would you know if the Java docs URL would be stable? 

If a link exists to stable content then referring to it would be ideal. That way, we could avoid maintaining two sources of information in the event that updates occur. 

Do you think listing the names of the attributes might be misleading?

Reply:
Yeah, that page is generated from the code, so I assume it would be up-to-date. Completely agree with you on avoiding duplication. I think we could only refer the users to the page then, without even listing the names, so we avoid duplicating stuff. What do you think?

Sorry for this adding and removing stuff.

Reply:
Hi @jabolina . Yeah, the link should provide context to each attribute. Is there a specific grouping for these attributes types? `StateTransferManager` not OK to use?

How about?

> A state transfer exposes attributes as properties, so that {brandname} can export metrics from each cache

Reply:
Not that I am aware of. The grouping only exists when using JMX to query the values, other than that, they are not grouped together. That's why I think we should avoid adding the name in the listing because this is related to JMX. I think we are good to go with lines 6 and 7, and the additional resources give more context.

Reply:
Hi @jabolina .

I removed the list and added:

> A state transfer exposes attributes as properties, so that {brandname} can export metrics from each cache.

By mentioning _attributes_ a link relationship exists with the link provided at the end of the module. 

Please let me know if your happy with me to progress to the next review stage?

Reply:
Hi @jabolina . Would you be OK to approve the PR?

**File changes** tab > **Review changes** drop-down menu > **Approved**